src/anim/controller/anim-transition.js

```diff
@@ -20,15 +18,14 @@ import {
  * @param {string} interruptionSource - Defines whether another transition can interrupt this one and which of the current or previous states transitions can do so. One of ANIM_INTERRUPTION_*.
  */
 class AnimTransition {
-    constructor(controller, from, to, time, priority, conditions, exitTime, transitionOffset, interruptionSource = ANIM_INTERRUPTION_NONE) {
-        this._controller = controller;
+    constructor({ from, to, time = 0, priority = 0, conditions = [], exitTime = null, transitionOffset = null, interruptionSource = ANIM_INTERRUPTION_NONE }) {
```

Comment:
As this is private, I'm less worried about this but the optional params defaults would need to be documented. Although saying that, would IDEs show the defaults?

Reply:
I wouldn't make this class public. Do IDE's show private jsdoc comments?

Reply:
They do when you start calling the function IIRC. Does depend on the IDE though. This is super minor though

Reply:
![image](https://user-images.githubusercontent.com/16639049/123076548-28dc7080-d411-11eb-9485-cc208d1858b0.png)

Reply:
Ok, I can add these in 👍